### PR TITLE
LOG-411 All pods are allocated to linux nodes

### PR DESCRIPTION
**Pull request by lukas-vlcek**
Replaces https://github.com/openshift/cluster-logging-operator/pull/198

### CODE REVIEW

File: pkg/k8shandler/util.go [hunk 1]
```diff
@@ -9,10 +9,34 @@ import (
...
 
+const (
+	OsNodeLabel = "kubernetes.io/os"
+	LinuxValue  = "linux"
+)
+
+// EnsureLinuxNodeSelector takes given selector map and returns a selector map with linux node selector added into it.
+// If there is already a node type selector and is different from "linux" then it is overridden and warning is logged.
+// See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#interlude-built-in-node-labels
+func EnsureLinuxNodeSelector(selectors map[string]string) map[string]string {
```
Comment (ewolinetz) [inline pkg/k8shandler/util.go]: golang uses capitalized function names to denote public vs private. I don't think we need to have this method be publicly available... can you make it `ensureLinuxNodeSelector` instead?

**Comment by ewolinetz** [comment]
small nit regarding function name, otherwise lgtm

File: pkg/k8shandler/util.go (hunk 1 above)
Comment (lukas-vlcek) [reply pkg/k8shandler/util.go]: Done. The original idea was that this function would be later called (re-used) from CLO as well. But we can make it public only once we get to that point in the future.

**Pull request merged**